core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/Compression.java

```diff
@@ -250,33 +277,71 @@ public void initializeDefaultCodec() {
        */
       @Override
       protected CompressionCodec createNewCodec(final int bufferSize) {
-        DefaultCodec myCodec = new DefaultCodec();
-        Configuration myConf = new Configuration(conf);
-        // only use the buffersize if > 0, otherwise we'll use
-        // the default defined within the codec
-        if (bufferSize > 0)
-          myConf.setInt(BUFFER_SIZE_OPT, bufferSize);
-        myCodec.setConf(myConf);
-        return myCodec;
+        if (USE_QAT) {
+          String extClazz =
+              (conf.get(CONF_QAT_CLASS) == null ? System.getProperty(CONF_QAT_CLASS) : null);
+          String clazz = (extClazz != null) ? extClazz : DEFAULT_QAT_CLASS;
```

Comment:
I'm a little confused by this logic. This reads:

If `io.compression.codec.qat.class` is not set in the Hadoop config file, then use the class name defined by `-Dorg.apache.hadoop.io.compress.QatCodec.class=????`. This part makes sense, although I don't know why we'd want to be able to override the QAT codec class name, since it seems like it would be a well-known value.

If it **is** set in the Hadoop config file, then just set the class name to `null`, which will result in us using the default class name we've hard-coded. This part doesn't make sense to me. Why wouldn't we use the class name from the Hadoop config file in this case?

Also, shouldn't we prioritize our explicit `-D` override instead of only using it when the Hadoop config doesn't set it? The implementation here will prioritize what's in the the default value if the Hadoop config is set, and ignore the override.

I think the logic should be:

1. Check if system property is set to non-null. Use it if set, otherwise
2. check if Hadoop config is set to non-null. Use it if set, otherwise
3. use the default class name that we've hard-coded.

You could also use the first one in that order where the class isn't just non-null, but also the class loads.

Also, minor nit: avoiding unnecessary negations in ternary statements helps readability:
```java
String clazz = extClazz == null ? DEFAULT_QAT_CLASS : extClazz;
```

Reply:
I was just doing what the others do here (mostly copy/paste). LZO does it [here](https://github.com/apache/accumulo/blob/1.10/core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/Compression.java#L159) and Snappy does it [here](https://github.com/apache/accumulo/blob/1.10/core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/Compression.java#L387)

Reply:
Oh wow. That existing code seems very broken. I'm fine with following suit with what the rest of the code does and fixing it later. It's especially concerning that we completely ignore the value in the Hadoop config when it's set there, and use our property when it's not set there. I'm not sure why we even bothered to read the value from the Hadoop config at all in that case, if it has no bearing on what we're going to do.